File: GameMarket-Server/Tables.cpp
#include"Tables.h"


namespace tables_namespace
{
	std::string toString(Tables collection)
	{
		switch (collection)
		{
			//xbox tables
		case Tables::XboxGameGenres: return "GameGenres";
		case Tables::XboxGameTitles: return "GameTitles";
		case Tables::XboxUserProfiles: return "UserProfiles";
		case Tables::XboxProductIds: return "ProductIds";
		case Tables::XboxGameBundles: return "GameBundles";
		case Tables::XboxTitleDetails: return "TitleDetails";
		case Tables::XboxMarketDetails: return "MarketDetails";
		case Tables::XboxGroupData: return "GroupData";
		
			//steam tables
		case Tables::SteamAppIDs: return "APPIds";
		case Tables::SteamAppDetails: return "AppDetails";
		case Tables::SteamAppPublishers: return "AppPublishers";
		case Tables::SteamAppDevelopers: return "AppDevelopers";
		case Tables::SteamAppGenres: return "AppGenres";
		case Tables::SteamPackageIDs:return "PackageIDs";
		case Tables::SteamPackageDetails: return "PackageDetails";
		case Tables::SteamPackages: return "Packages";
		case Tables::SteamAppPlatforms: return "AppPlatforms";

		default: return "";
		}
	}
	void XboxGameMarketData::outputData()
	{
		std::cout << "Product IDs:  ";
		for (std::string id : productIDs)
		{
			std::cout << id << " ";
		}
		std::cout << std::endl;
		std::cout << "ProductID: " << productID << std::endl;
		std::cout << "Dev Name: " << devName << std::endl << "Pub Name: " << pubName << std::endl << "Title: "
			<< productTitle << std::endl << "productDesc: " << std::endl << productDesc << std::endl;

		std::cout << "ListPrice: " << std::to_string(listPrice) << std::endl << "MSRP:  " << std::to_string(MSRP) << std::endl;
	}

	void XboxGameTitleData::outputData()
	{
		cout << "Title: " << titleName << endl;
		cout << "TitleId: " << titleID << endl;
		cout << "ModernTitleId: " << modernTitleID << endl;
		cout << "Game Pass: " << isGamePass << endl;
		cout << "Devices: ";
		for (string device : devices)
		{
			cout << device << ", ";
		}
		cout << endl <<endl;
	}
	void XboxTitleDetailsData::outputData()
	{
		cout << "titleID: " << titleID << endl;
		cout << "productID: " << productID << endl;

		cout << "Bundle IDs: ";
		for (string bundle : bundleIDs)
		{
			cout << bundle << "  ";
		}
		cout << endl << endl;
	}
	void SteamAPPListData::outputData()
	{
		cout << "APPid: " << appid << endl;
		cout << "Name: " << name << endl <<endl;
	}
}

File: GameMarketAPIServer/Models/Enums/Tables.cs
using GameMarketAPIServer.Services;
using static GameMarketAPIServer.Services.DataBaseManager;

namespace GameMarketAPIServer.Models.Enums
{
    public enum CRUD
    {
        Create,
        Read,
        Update,
        Delete,
        UpSert
    };
    public enum Tables
    {
        //xbox
        XboxUserProfiles,
        XboxGameTitles,
        XboxGameBundles,
        XboxProductIds,
        XboxTitleDetails,
        XboxMarketDetails,
        XboxGameGenres,
        XboxGroupData,
        XboxTitleDevices,
        XboxProductPlatforms,


        //steam
        SteamAppIDs,
        SteamAppDetails,
        SteamAppDevelopers,
        SteamAppPublishers,
        SteamPackages,
        SteamPackageDetails,
        SteamPackageIDs,
        SteamAppPlatforms,
        SteamAppGenres,


        //gamemarket
        GameMarketGameTitles,
        GameMarketXboxLink,
        GameMarketSteamLink,
        GameMarketPublishers,
        GameMarketDevelopers,
        //other
    };


    public static class TablesExtension
    {
        public static string To_String(this Tables table)
        {
            switch (table)
            {

                //xbox tables
                case Tables.XboxGameGenres: return "GameGenres";
                case Tables.XboxGameTitles: return "GameTitles";
                case Tables.XboxUserProfiles: return "UserProfiles";
                case Tables.XboxProductIds: return "ProductIds";
                case Tables.XboxGameBundles: return "GameBundles";
                case Tables.XboxTitleDetails: return "TitleDetails";
                case Tables.XboxMarketDetails: return "MarketDetails";
                case Tables.XboxGroupData: return "GroupData";
                case Tables.XboxTitleDevices: return "titleDevices";
                case Tables.XboxProductPlatforms: return "productPlatforms";

                //steam tables
                case Tables.SteamAppIDs: return "appIDs";
                case Tables.SteamAppDetails: return "AppDetails";
                case Tables.SteamAppPublishers: return "AppPublishers";
                case Tables.SteamAppDevelopers: return "AppDevelopers";
                case Tables.SteamAppGenres: return "AppGenres";
                case Tables.SteamPackageIDs: return "PackageIDs";
                case Tables.SteamPackageDetails: return "PackageDetails";
                case Tables.SteamPackages: return "Packages";
                case Tables.SteamAppPlatforms: return "AppPlatforms";


                case Tables.GameMarketGameTitles: return "GameTitles";
                case Tables.GameMarketXboxLink: return "XboxLink";
                case Tables.GameMarketSteamLink: return "SteamLink";
                case Tables.GameMarketDevelopers: return "Developers";
                case Tables.GameMarketPublishers: return "Publishers";

                default: return "";
            }
        }


        public static DataBaseManager.Schemas ToSchema(this Tables table)
        {
            string tableEnum = table.ToString().ToLower();

            if (tableEnum.Contains("xbox")) return Schemas.xbox;
            else if (tableEnum.Contains("steam")) return Schemas.steam;
            else if (tableEnum.Contains("gamemarket")) return Schemas.gamemarket;

            else 
                return DataBaseManager.Schemas.xbox;
        }
        public static int toInt(this Tables table)
        {

            return (int)table;
        }
    }
}
